packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -51,21 +65,59 @@ export class SharedTreeCore<TChangeRebaser extends ChangeRebaser<any, any, any>>
         this.rebaser = new Rebaser(changeRebaser);
         this.localState = this.rebaser.empty;
         this.sequencedRevision = this.rebaser.empty;
+
+        this.summaryElements = indexes.map((i) => i.summaryElement).filter((e): e is SummaryElement => e !== undefined);
+        assert(
+            new Set(this.summaryElements.map((e) => e.key)).size === this.summaryElements.length,
+            "Index summary element keys are not unique",
+        );
     }
 
     // TODO: SharedObject's merging of the two summary methods into summarizeCore is not what we want here:
     // We might want to not subclass it, or override/reimplement most of its functionality.
     protected summarizeCore(serializer: IFluidSerializer, telemetryContext?: ITelemetryContext): ISummaryTreeWithStats {
-        // TODO: Do something like this loop for most of the methods in here.
-        for (const index of this.indexes) {
-            index.summaryElement?.getAttachSummary((contents) => this.serializer.stringify(contents, this.handle));
+        let stats = mergeStats();
+        const summary: ISummaryTree = {
+            type: SummaryType.Tree,
+            tree: {},
+        };
+        stats.treeNodeCount += 1;
+
+        // Merge the summaries of all indexes together under a single ISummaryTree
+        if (this.summaryElements.length > 0) {
```

Comment:
Is there a reason to have this "if"?

handling the empty case different from the non-empty case seems like an unneeded edge case (ex: could complicate the summary load logic slightly, add an extra thing to test etc.)

Reply:
My thought was that we don't need to put an "indexes" tree in the summary if there are no indexes. 

```
{
	indexes: {}
}
```

It's kind of like having a trait with nothing in it; it may as well not be there. However you are right that it means code loading it has to check for its existence before reading it, which is another thing to mess up. I don't have a strong opinion here. Since it's such a small thing that it's not going to bloat summary size or anything, I'm leaning towards removing the `if` as you suggest.

Reply:
Removed